pkg/api/validation/validation.go

```diff
@@ -1242,6 +1242,29 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldForbidden("data", "Maximum secret size exceeded"))
 	}
 
+	switch secret.Type {
+	case api.SecretTypeOpaque, "":
+		// no-op
+	default:
+		allErrs = append(allErrs, errs.NewFieldInvalid("type", secret.Type, "unknown SecretType"))
```

Comment:
@smarterclayton @liggitt Honestly, I don't feel like types are well-thought out yet.  I have a feeling that docker/rkt pull with secret will help us work through them.  I'm fine with this constraint for now.